lib/predicate/requestWrapper.js

```diff
@@ -0,0 +1,114 @@
+import stream from 'stream';
+import crypto from 'crypto';
+
+const reqSym = Symbol('req');
+const bufSym = Symbol('buf');
+const outputSym = Symbol('output');
+
```

Comment:
I blew it here. Thanks for testing this one :) Turns out the file I was using was too small to trigger a second `read()` event. I've updated [requestWrapper](https://github.com/austo/S3/blob/master/lib/predicate/requestWrapper.js#L68) to be more idiomatic and have tested the change against your `changeContent` code along with a few local tests I didn't include in the PR.